README.md

```diff
@@ -2,34 +2,47 @@
 Pipeline for running single cell ENCODE E2G.
 
 Input: Multiome dataset (ATAC and RNA)
-Output: Enhancer -> Gene Predictions
+Output: Enhancer -> Gene Predictions (for each cell cluster)
 
 The pipeline consists of the following components:
-1. Compute ABC model predictions
+1. Compute ABC model predictions for each cell cluster
 2. Generate E2G features from ABC predictions
-3. Compute Kendall Correlation
-5. Combine 2 & 3 to construct a feature file to be used as input to the ML model
-6. Have ML model make predictions (assigning a score to each E-G pair)
+3. Compute Kendall correlation and/or ARC-E2G score for each cell cluster
+4. Combine 2 & 3 to construct a feature file to be used as input to predictive model
+5. (optional) Train predictive model using CRISPR-validated E-G pairs from K562 dataset
+6. Apply trained model to make predictions by assigning a score to each E-G pair
 
-## Usage
+## Set up
 Clone the repo and set it up for submdoule usage
 ```
 git clone --recurse-submodules https://github.com/EngreitzLab/sc-E2G.git
 git config --global submodule.recurse true
 ```
 
-Modify `config/config_biosamples.tsv` with your multiome data
-
 When running for the first time, the conda environments have to be setup.
 For speed, it's recommended that your current environment has mamba installed
 
 ```
 conda config --set channel_priority flexible  # Make sure your conda config uses flexible channel packaging to prevent unsatisfiable errors
-conda create -n mamba -c conda-forge mamba
+conda create -n mamba -c conda-forge mamba python=3.11
 conda activate mamba
-mamba install -c conda-forge -c bioconda snakemake
+mamba install -c conda-forge -c bioconda snakemake=7
 ```
 
+## Apply model
+Before running this workflow, users should perform clustering to define cell clusters through regular single-cell analysis (such as Seurat & Signac).
+Required input data includes (refer to the example data in the `example_chr22_multiome` folder):
+1. Combined meta_data, including columns for cell_name, sample, cluster, and barcode.
+2. Combined RNA matrix, normalized by the total UMI count of each cell. It can be with or without log transformation.
+3. Indexed fragment files for each sample.
+
+To configure the pipeline:
+- Modify `config/config.yaml` to specify paths for meta_data, rna_matrix, and results_dir.
+- Modify `config/config_multiome_samples.tsv` to specify the fragment file path for each sample.
+- Modify `config/config_cell_clusters.tsv` to specify the Hi-C file path, Hi-C data type, Hi-C resolution, TSS coordinates, and gene coordinates for each cell cluster.
```

Comment:
let's discuss how we want to structure and specify the configs (eg. what is the relationship between config_multiome_samples, config_cell_clusters, and config_dataset?)

Reply:
From what I understand, config_cell_clusters serves the same purpose of dataset_config (both of the type "ABC biosample configs" and config_multiome_samples provides the fragment file that needs to be split into clusters? So "samples" = "datasets" and cell clusters are within a sample, and we ultimately want to compute predicitons on the cell cluster level. We could potentially rename some columns to make this clear (eg. sample --> dataset, cell_cluster --> biosample or something). Won't there also be an RNA matrix and metadata per cluster (or dataset/sample?), rather than one for the overall run? These could be additional columns.